iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/VoltageRegulationImpl.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.impl.extensions;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.util.trove.TBooleanArrayList;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.VoltageRegulation;
+import com.powsybl.iidm.network.impl.AbstractMultiVariantIdentifiableExtension;
+import gnu.trove.list.array.TDoubleArrayList;
+
+/**
+ * @author Coline Piloquet {@literal <coline.piloquet@rte-france.fr>}
+ */
+public class VoltageRegulationImpl extends AbstractMultiVariantIdentifiableExtension<Battery> implements VoltageRegulation {
+
+    private final TBooleanArrayList voltageRegulatorOn;
+
+    private final TDoubleArrayList targetV;
+
+    private Terminal regulatingTerminal;
+
+    public VoltageRegulationImpl(Battery battery, Terminal regulatingTerminal, Boolean voltageRegulatorOn, double targetV) {
+        super(battery);
+        checkRegulatingTerminal(regulatingTerminal, getNetworkFromExtendable());
+        if (voltageRegulatorOn == null) {
+            throw new PowsyblException("Voltage regulator status is not defined");
+        }
+        int variantArraySize = getVariantManagerHolder().getVariantManager().getVariantArraySize();
+        this.setRegulatingTerminal(regulatingTerminal);
+        this.voltageRegulatorOn = new TBooleanArrayList(variantArraySize);
+        this.targetV = new TDoubleArrayList(variantArraySize);
+        for (int i = 0; i < variantArraySize; i++) {
+            this.voltageRegulatorOn.add(voltageRegulatorOn);
+            this.targetV.add(targetV);
+        }
+    }
+
+    private static void checkRegulatingTerminal(Terminal regulatingTerminal, Network network) {
+        if (regulatingTerminal != null && regulatingTerminal.getVoltageLevel().getNetwork() != network) {
+            throw new PowsyblException("regulating terminal is not part of the same network");
+        }
+    }
+
+    @Override
+    public Terminal getRegulatingTerminal() {
+        return regulatingTerminal;
+    }
+
+    @Override
+    public void setRegulatingTerminal(Terminal regulatingTerminal) {
```

Comment:
Just thinking at the behaviour of the SlackTerminal extension... It's strange to support multivariant and have a setter here but not supporting multivariant for that parameter.

Reply:
The regulating terminal is not variant dependent...